patroni/postgresql/slots.py

```diff
@@ -1,37 +1,85 @@
 import logging
+import os
 
-from patroni.postgresql.connection import get_connection_cursor
 from collections import defaultdict
+from contextlib import contextmanager
+
+from .connection import get_connection_cursor
+from .misc import format_lsn
 
 logger = logging.getLogger(__name__)
 
 
-def compare_slots(s1, s2):
+def compare_slots(s1, s2, dbid='database'):
     return s1['type'] == s2['type'] and (s1['type'] == 'physical' or
```

Comment:
which case does the second part of this condition cover (the `and (s1['type'] == 'physical' or ...` part )?

Reply:
The slot type could be either `physical` or `logical`.